frontend/lib/src/theme/utils.ts

```diff
@@ -417,6 +417,27 @@ export function hasLightBackgroundColor(theme: EmotionTheme): boolean {
   return getLuminance(theme.colors.bgColor) > 0.5
 }
 
+export function getTextColors(theme: EmotionTheme): any {
+  const red = getMdRed(theme)
```

Comment:
nit: I'm a bit worried that we are calling the `hasLightBackgroundColor` an immense amount of times since every color repeats this check. I have no clue if the luminance check requires any significant enough computation that would end up slowing down Streamlit (probably not), but maybe we can unify this into one method that returns all colors and removes all the individual color methods, e.g.:

```
const hasLightBackground = hasLightBackgroundColor(theme)
const red = hasLightBackground ? theme.colors.red80 : theme.colors.red70
...
```

Reply:
Makes sense, cleaned it up 👍🏼